Assets/MRTK/SDK/Editor/Inspectors/UX/InteractiveElement/TrackedStatesInspector.cs

```diff
@@ -0,0 +1,282 @@
+
+            RenderStates();
+
+            EditorGUILayout.Space();
+            EditorGUILayout.Space();
+
+            RenderAddCoreStateButton();
+
+            EditorGUILayout.Space();
+            EditorGUILayout.Space();
+
+            RenderCreateNewStateButton();
+
+            serializedObject.ApplyModifiedProperties();
+        }
+
+        private void RenderStates()
+        {
+            for (int i = 0; i < stateList.arraySize; i++)
+            {
+                using (new EditorGUILayout.VerticalScope(EditorStyles.helpBox))
+                {
+                    SerializedProperty state = stateList.GetArrayElementAtIndex(i);
+                    SerializedProperty stateName = state.FindPropertyRelative("stateName");
+                    SerializedProperty stateValue = state.FindPropertyRelative("stateValue");
+                    SerializedProperty stateEventConfiguration = state.FindPropertyRelative("eventConfiguration");
+
+                    EditorGUILayout.Space();
+
+                    using (new EditorGUILayout.HorizontalScope())
+                    {
+                        InspectorUIUtility.DrawLabel(stateName.stringValue, 14, InspectorUIUtility.ColorTint10);
+
+                        // Draw the state value in the state while the application is playing for debugging
+                        if (Application.isPlaying)
+                        {
+                            EditorGUILayout.LabelField(stateValue.intValue.ToString());
+                        }
+
+                        // Do not draw a remove button for the default state
+                        if (stateName.stringValue != "Default")
+                        {
+                            // Draw a button with a '-' for state removal
+                            if (InspectorUIUtility.SmallButton(RemoveStateButtonLabel))
+                            {
+                                stateList.DeleteArrayElementAtIndex(i);
+                                break;
+                            }
+                        }
+                    }
+
+                    using (new EditorGUILayout.VerticalScope())
+                    {
+                        EditorGUILayout.Space();
+                        EditorGUILayout.Space();
+
+                        using (new EditorGUI.IndentLevelScope())
+                        {
+                            // If an event configuration for a state exists then draw the event config scriptable 
+                            RenderStateEventConfiguration(stateName, stateEventConfiguration);
+
+                            // When a new core state is added via inspector, the name is initialized to "New Core State" and then changed
+                            // to the name the user selects from the enum list of CoreInteractionStates
+                            if (stateName.stringValue == "New Core State")
+                            {
+                                SetCoreStateType(state, stateName);
```

Comment:
It would be very nice if CoreInteractionStates was not a hard coded enum, since then users could define their own states as needed. In current implementation, adding a new state would require a change to MRTK.

Reply:
A user can create their own state.  The enum is there to communicate which states have already defined state setting logic.  If a user decides to create their own state like this:

![CreateNewState](https://user-images.githubusercontent.com/53493796/92526371-3a35f280-f1da-11ea-9108-2c269f11d0e5.gif)

or via script like this:

`interactiveElement.AddNewState("MyNewState")`

Then they add their own script with `interactiveElement.SetStateOn("MyNewState")`.  Adding a new state does not require a change to MRTK.